dev_guide/builds/advanced_build_operations.adoc

```diff
@@ -122,3 +122,123 @@ endif::[]
 If the specified `NodeSelector` cannot be matched to a node with those labels,
 the build still stay in the `Pending` state indefinitely.
 ====
+
+[[dev-guide-chaining-builds]]
+== Chaining Builds
+
+For compiled languages (Go, C, C++, Java, etc.) the dependencies necessary for compilation might increase the size of the
+image or introduce vulnerabilities that can be exploited.
+
+To avoid these problems, a
+xref:../../architecture/core_concepts/builds_and_image_streams.adoc#source-build[Source-to-Image]
+build can be chained with a
+xref:../../architecture/core_concepts/builds_and_image_streams.adoc#docker-build[Docker]
+build in such a way that one image with all the tools necessary for compilation is used to compile the artifact,
+while a separate image is used to host and execute the compiled artifact.
+
+[ditaa, "chained-build"]
+----
+
+ +---------+
+ |   S2I   |       +----------+
+ | Builder |   +   |  Source  |
+ |  Image  |       +----+-----+
+ +---------+            |
+      |                 |
+      |  +-----------+  |
+      +->| S2I Build |<-+
+         +-----+-----+
+               |
+               v
+       +---------------+       +--------------+
+       |   Image with  |       |    Runtime   |        +--------------+
+       |     Binary    |   +   |     Image    |   +    |  Dockerfile  |
+       |    Artifact   |       +-------+------+        +------+-------+
+       +-------+-------+               |                      |
+               |                       |                      |
+               |                       v                      |
+               |               +--------------+               |
+               |               | Docker Build |               |
+               +-------------->|     with     |<--------------+
+                               | Image Source |
+                               +--------------+
+                                       |
+                                       |
+                                       v
+                               +---------------+
+                               | Runtime Image |
+                               | with Compiled |
+                               |    Artifact   |
+                               +---------------+
+----
+
+The first build, which takes source and produces a binary artifact, is a a regular Source-to-Image
+build. In the following example, a JEE source repository is to build a WAR file that is output to
+the `artifact-image` ImageStream. The path of the output artifact will depend on the assemble script
+of the Source-to-Image builder used. In this case, it will be output to `/wildfly/standalone/deployments/ROOT.war`.
+
+[source,yaml]
+----
+apiVersion: v1
+kind: BuildConfig
+metadata:
+  name: artifact-build
+spec:
+  output:
+    to:
+      kind: ImageStreamTag
+      name: artifact-image
+  source:
+    git:
+      uri: https://github.com/bparees/openshift-jee-sample.git
```

Comment:
i think it's time to put a copy of this repo in the openshift org, i don't need my name in the official product docs :)